src/main/scala/com/cognite/sdk/scala/v1/views.scala

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 Cognite AS
+// SPDX-License-Identifier: Apache-2.0
+
+package com.cognite.sdk.scala.v1
+import com.cognite.sdk.scala.common.DomainSpecificLanguageFilter
+import com.cognite.sdk.scala.v1.containers.{ContainerPropertyType, ContainerReference}
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+
+final case class ViewReference(
+    space: String,
+    externalId: String,
+    version: Option[String] = None
+) {
+  val `type`: String = "view"
+}
+object ViewReference {
+  implicit val viewReferenceEncoder: Encoder[ViewReference] =
+    deriveEncoder[ViewReference]
```

Comment:
Are you sure that you don't need to use `type` in `Encoder[ViewReference]`? Auto derived encoder will not include `type` 🤔

Reply:
hmm right, i will copy your eencoder